lib/Optimizer/Lower.cpp

```diff
@@ -357,7 +357,7 @@ static void lowerSGDNode(Function *F, CompilationContext &cctx,
   // OptimizationStochasticGradientDescent/
   if (momentum > 0.0) {
     Placeholder *Gsum =
-        F->getParent()->createPlaceholder(W.getType(), "gsum", false);
+        F->getParent()->createPlaceholder(W.getType(), "gsum", true);
```

Comment:
This needs to be initialized to zero but it is not trainable. It is unlikely someone would ever initialize it in another way because `gsum` is supposed to be a geometric series of previous weight updates with `momentum` as the ratio between successive terms.

@putivsky ran into something similar during his work on loading models for training with the mean/variance inputs of `SpatialBN`. He might have some ideas on the best fix.

Reply:
I mean the only other option is to add another flag to Placeholder. I started there but we only used it in one place so I figured I'd try to reduce extra code. Not super fussed though.

Reply:
We use `Placeholder::isTraining` during differentiation to determine which nodes should have SGD nodes and weight updates added. That didn't matter here because the fact that an `SGDNode` is being lowered means that the graph has already been differentiated.